Support missing SNIMissingWarning in tests

pquentin opened the pull request:
urllib3 2.0 only supports OpenSSL 1.1.1+ which means SNI is mandatory, and removed the warning altogether.

nateprewitt reviewed (approved):
Fair enough, this looks good. Thanks @pquentin!

nateprewitt commented:
Looks like there's a small linting issue. Could you update the PR using our pre-commit hooks?

Although where it's moving the test import doesn't look right to me 👀. I'll investigate that separately.

pquentin commented:
Oops, sorry about that. Fixed the lint.

> Although where it's moving the test import doesn't look right to me 👀. I'll investigate that separately.

This is because I used `from tests import X` instead of `from . import X` and isort does not know that `tests` is a [known first party](https://pycqa.github.io/isort/docs/configuration/options.html#known-first-party). I switched back to `from . import X`.
